test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.cs

```diff
@@ -99,13 +100,68 @@ public class HubConnectionTests
         }
 
         [Fact]
+        public async Task CancelStartAsync()
+        {
+            var connection = new TestConnection();
+            var hubConnection = new HubConnection(connection, new JsonHubProtocol(), new LoggerFactory());
+
+            var token = new CancellationToken(true);
+            var startTask = hubConnection.StartAsync(token);
+
+            await Assert.ThrowsAsync<OperationCanceledException>(async () => await startTask);
+
+            await hubConnection.DisposeAsync();
+        }
+
+        [Fact]
+        public async Task CancelInvokeAsync()
+        {
+            var connection = new TestConnection();
+            var hubConnection = new HubConnection(connection, new JsonHubProtocol(), new LoggerFactory());
+
+            var token = new CancellationToken(true);
+            await hubConnection.StartAsync().OrTimeout();
+            var invokeTask = hubConnection.InvokeAsync<int>("testMethod", token);
+
+            await Assert.ThrowsAsync<TaskCanceledException>(async () => await invokeTask);
```

Comment:
You should make sure that the connection is still usable after this happens. A single invocation was cancelled but it should not break the connection.